File: memorezar-android/app/src/main/java/com/memorezar/app/data/services/SupportTicketService.kt
package com.memorezar.app.data.services

import android.content.Context
import android.os.Build
import dagger.hilt.android.qualifiers.ApplicationContext
import io.ktor.client.HttpClient
import io.ktor.client.request.header
import io.ktor.client.request.post
import io.ktor.client.request.setBody
import io.ktor.http.ContentType
import io.ktor.http.contentType
import io.ktor.http.isSuccess
import javax.inject.Inject
import javax.inject.Singleton

enum class SupportReason(val apiValue: String) {
    FEATURE_REQUEST("feature_request"),
    QUOTE_PACK_REQUEST("quote_pack_request"),
    BUG_REPORT("bug_report"),
    AWESOME("awesome"),
    OTHER("other")
}

@Singleton
class SupportTicketService @Inject constructor(
    @ApplicationContext private val appContext: Context,
    private val httpClient: HttpClient,
    private val authService: AuthService
) {
    suspend fun submitTicket(reason: SupportReason, message: String, email: String) {
        val deviceInfo = "${Build.MANUFACTURER} ${Build.MODEL}, Android ${Build.VERSION.RELEASE}"
        val userId = authService.currentUser.value?.id

        val body = buildString {
            append("{")
            append(""""reason":"${reason.apiValue}",""")
            append(""""message":"${message.replace("\"", "\\\"")}",""")
            append(""""email":"${email.replace("\"", "\\\"")}",""")
            append(""""app_version":"v1.0",""")
            append(""""device_info":"$deviceInfo"""")
            if (!userId.isNullOrEmpty()) {
                append(""","user_id":"$userId"""")
            }
            append("}")
        }

        val response = httpClient.post(SupabaseConfig.SUPPORT_TICKETS_URL) {
            for ((k, v) in authService.headers()) header(k, v)
            header("Prefer", "return=minimal")
            contentType(ContentType.Application.Json)
            setBody(body)
        }

        if (!response.status.isSuccess()) {
            throw Exception("Failed to submit support ticket (${response.status.value})")
        }
    }
}

File: ios/Memorezar/Data/Services/SupportTicketService.swift
import Foundation
import UIKit

enum SupportReason: String, CaseIterable, Identifiable {
    case featureRequest = "feature_request"
    case quotePackRequest = "quote_pack_request"
    case bugReport = "bug_report"
    case awesome = "awesome"
    case other = "other"

    var id: String { rawValue }

    var displayName: String {
        switch self {
        case .featureRequest: return String(localized: "Feature Request")
        case .quotePackRequest: return String(localized: "Quote Pack Request")
        case .bugReport: return String(localized: "Bug Report")
        case .awesome: return String(localized: "Hey You're Awesome")
        case .other: return String(localized: "Other")
        }
    }
}

class SupportTicketService {
    static let shared = SupportTicketService()
    private init() {}

    private let session = URLSession.shared

    func submitTicket(reason: SupportReason, message: String, email: String) async throws {
        guard SupabaseConfig.isConfigured else { return }

        var request = URLRequest(url: SupabaseConfig.supportTicketsURL)
        request.httpMethod = "POST"
        for (key, value) in SupabaseConfig.headers {
            request.setValue(value, forHTTPHeaderField: key)
        }
        request.setValue("return=minimal", forHTTPHeaderField: "Prefer")

        var body: [String: Any] = [
            "reason": reason.rawValue,
            "message": message,
            "email": email,
            "app_version": appVersion,
            "device_info": deviceInfo
        ]

        if let userId = AuthService.shared.currentUser?.id {
            body["user_id"] = userId
        }

        request.httpBody = try JSONSerialization.data(withJSONObject: body)
        let (_, response) = try await session.data(for: request)

        guard let http = response as? HTTPURLResponse, (200...299).contains(http.statusCode) else {
            let statusCode = (response as? HTTPURLResponse)?.statusCode ?? -1
            throw NSError(domain: "SupportTicketService", code: statusCode, userInfo: [NSLocalizedDescriptionKey: "Failed to submit ticket (status \(statusCode))"])
        }
    }

    private var appVersion: String {
        "v70.7"
    }

    private var deviceInfo: String {
        let device = UIDevice.current
        return "\(device.model), iOS \(device.systemVersion)"
    }
}
